rs/nervous_system/tools/release-runscript/src/main.rs

```diff
@@ -1,16 +1,92 @@
+use anyhow::{bail, Result};
+use clap::{Parser, Subcommand};
 use colored::*;
 use std::io::{self, Write};
 
-struct Step {
-    title: &'static str,
-    description: &'static str,
+#[derive(Debug, Parser)]
+struct DetermineTargets;
+
+#[derive(Debug, Parser)]
+struct RunTests;
+
+#[derive(Debug, Parser)]
+struct CreateProposalTexts;
+
+#[derive(Debug, Parser)]
+struct SubmitProposals;
+
+#[derive(Debug, Parser)]
+struct CreateForumPost;
+
+#[derive(Debug, Parser)]
+struct ScheduleVote;
+
+#[derive(Debug, Parser)]
+struct UpdateCanistersJson;
+
+#[derive(Debug, Parser)]
+struct UpdateChangelog;
+
+#[derive(Debug, Subcommand)]
+enum Step {
+    #[command(about = "Step 2: Determine Upgrade Targets")]
+    DetermineTargets(DetermineTargets),
+    #[command(about = "Step 3: Run NNS Upgrade Tests")]
+    RunTests(RunTests),
+    #[command(about = "Step 4: Create Proposal Texts")]
+    CreateProposalTexts(CreateProposalTexts),
+    #[command(about = "Step 5: Submit Proposals")]
+    SubmitProposals(SubmitProposals),
+    #[command(about = "Step 6: Create Forum Post")]
+    CreateForumPost(CreateForumPost),
+    #[command(about = "Step 7: Schedule Trusted Neurons Vote")]
+    ScheduleVote(ScheduleVote),
+    #[command(about = "Step 8: Update Mainnet Canisters")]
+    UpdateCanistersJson(UpdateCanistersJson),
+    #[command(about = "Step 9: Update Changelog")]
+    UpdateChangelog(UpdateChangelog),
 }
 
-fn main() {
-    let steps = vec![
-        Step {
-            title: "Pick Release Candidate Commit",
-            description: "Run `./testnet/tools/nns-tools/cmd.sh latest_commit_with_prebuilt_artifacts`.
+#[derive(Debug, Parser)]
+#[clap(
+    name = "release-runscript",
+    about = "Release NNS and SNS canisters.",
+    version
+)]
+struct ReleaseRunscript {
+    #[command(subcommand)]
+    step: Option<Step>,
+}
+
+fn main() -> Result<()> {
+    let args = match ReleaseRunscript::try_parse_from(std::env::args()) {
+        Ok(args) => args,
+        Err(e) => {
+            bail!("{}", e);
+        }
+    };
+
+    print_header();
+
+    match args.step {
+        None => run_pick_commit(),
```

Comment:
in addition, make an explicit support for this step (and then you could maybe also use clap default value for "step")